flarefly/fitter.py

```diff
@@ -935,7 +948,7 @@ def func_mult(params):
                     floating=not self._fix_bkg_pars_[ipdf]['frac'])
 
         self._total_pdf_ = zfit.pdf.SumPDF(self._signal_pdf_+self._refl_pdf_+self._background_pdf_,
-                                           self._fracs_)
+                                           self._fracs_).to_truncated(limits = self._limits_, obs = obs, norm = obs)
```

Comment:
Debugging locally I found that the issue with MacOs python3.11 is that this SumPDF is built from the same PDFs as the one used to build the total binned pdf. Since we do not support the truncated PDFs in case of binned fits for the time being, I would suggest the following:
```suggestion
                                           self._fracs_)

        if not self._data_handler_.get_is_binned():
            self._total_pdf_ = self._total_pdf_.to_truncated(limits = self._limits_, obs = obs, norm = obs)
```
combined with the next suggestion.

The tests work in this case, I am not sure if there is any corner case where it breaks though.